patches/tModLoader/Terraria/Item.TML.cs

```diff
@@ -21,7 +21,12 @@ public partial class Item : TagSerializable
 		/// <summary>
 		/// The damage type of this Item. Assign to DamageClass.Melee/Ranged/Magic/Summon/Throwing, or ModContent.GetInstance<T>() for custom damage types.
 		/// </summary>
-		public DamageClass DamageType { get; set; }
+		public DamageClass DamageType;
+
+		/// <summary>
+		/// Set this to true to prevent this weapon or ammo item from being adjusted by damage modifiers.
+		/// </summary>
+		public bool IgnoreDamageModifiers;
```

Comment:
NoScaling means it has no class definition. IgnoreDamageModifiers simply means the weapon ignores damage modifiers. as such, the intention is to make it so that e.g. you can have a melee weapon that benefits from melee effects, but still doesn't benefit from any damage modifiers

Reply:
If someone wants to make a melee weapon that "benefits from melee effects, but still doesn't benefit from any damage modifiers", they can make a new damage class that counts as melee, but doesn't have scaling.